### PR TITLE
[Feature]if fe rollback version, support unmerged catalog types

### PR DESCRIPTION
**Pull request by adzfolc**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|2|2|100.00%|[]|

**Review by stephen-shelby — COMMENTED** [review]
LGTM

**Comment by stephen-shelby** [comment]
@gengjun-git PTAL

File: fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java [hunk 1]
```diff
@@ -132,6 +133,9 @@ public void replayCreateCatalog(Catalog catalog) throws DdlException {
         if (Strings.isNullOrEmpty(type)) {
             throw new DdlException("Missing properties 'type'");
         }
+        if (Config.ignore_unknown_connector_type && !CreateCatalogStmt.supportedCatalog.contains(type)) {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java:136]: The config param is not required. If encountered an unknown type, just return and print a warning log. Too many config params will mislead users.

**Review by gengjun-git — DISMISSED** [review]

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|3|3|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by adzfolc** [comment]
run starrocks_admit_test

**Comment by adzfolc** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**